src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,60 @@
+import { Button, LoadingSpinner } from '@components';
+import { AiOutlineClose } from 'react-icons/ai';
+import { ModalState } from '@constants';
+import { ReactNode } from 'react';
+import styles from './Modal.module.less';
+import { useDelayedMount } from 'util/delayed-mount/delayed-mount';
+
+interface Props {
+  ariaLabel: string,
+  state: ModalState,
+  setState: (state: ModalState) => void,
+  headerString: string,
+  body: ReactNode,
+  footer?: ReactNode
+}
+
+const Modal = (props: Props) => {
+  const { mounted, visible } = useDelayedMount(props.state !== ModalState.CLOSED, 200);
```

Comment:
I don't clearly see what happens here and why. Can you clarify what is the goal of `useDelayedMount` here?

Reply:
Yeah we talked about it, this is to delay the unmount of modal, to show the smooth opacity animation before it's gone from the dom